java/arcs/core/analysis/DependencyNode.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.analysis
+
+import arcs.core.data.AccessPath
+import arcs.core.data.expression.Expression
+
+/** Field [Identifier]. */
+private typealias Identifier = String
+
+/** Lists of [Identifier]s imply an [AccessPath]. */
+private typealias Path = List<Identifier>
+
+/**
+ * [DependencyNode]s represent how inputs of a Paxel [Expression] contribute to its outputs.
+ *
+ * A [DependencyNode] makes up a Directed-Acyclic-Graph to map fields of an output handle
+ * connection to fields in other handle connections.
```

Comment:
The is not very precise or clear. A graph as whole (and not the node) is the one that captures how the inputs contribute to output.